api/python/provisioner/commands/deploy.py

```diff
@@ -307,14 +307,12 @@ def _run_states(self, states_group: str, run_args: run_args_type):
                     "ha.cortx-ha"
                 ):
                     # Execute first on primary then on secondaries.
+                    if state == "ha.cortx-ha":
+                        self.ensure_consul_running()
                     self._apply_state(f"components.{state}", primary, stages)
                     self._apply_state(
                         f"components.{state}", secondaries, stages
                     )
-                else:
```

Comment:
We need else.
If it doesn't get filtered in any conditionals above, it still has to be executed for generic case.

Reply:
Ok.

Reply:
addressed